app/src/views/private/components/module-bar.vue

```diff
@@ -9,8 +9,8 @@
 				v-tooltip.right="modulePart.name"
 				icon
 				x-large
-				:to="modulePart.to"
-				:href="modulePart.url"
+				:to="modulePart.to || (modulePart.url && modulePart.url.startsWith('/') ? modulePart.url : undefined)"
+				:href="modulePart.url && modulePart.url.startsWith('/') ? undefined : modulePart.url"
```

Comment:
I think we can simplify the inline template logic quite a bit by moving this magic down to line 86, where we're creating the "to" field in the first place. By omitting "url" and "to" from the original object, and only adding it when it matches the condition, we can clean this up nicely 🙂

Reply:
Hmm, I did it in this manner as there was a need to modify `SettingsModuleBarLink` typings. Somehow I was wondering where's the typing for `modulePart.color`.

https://github.com/directus/directus/blob/f5550af862641ba81b3d8b96fe8210e30a59bd5c/app/src/views/private/components/module-bar.vue#L15-L20

Reply:
And updating the typings to include `to` also meant that there was a need to update `system-modules.vue` as the modulePart now will not have a `url` if it is changed to a `to`.

Reply:
Hmm, I was thinking whether to have separate typings for internal and external links. 

But when I do this, vetur will throw error on either `module-bar` or `system-module` depending on which variable is set to null. Just random thoughts on whether typings should be changed, so that people developing modules will be prompted to set either `url` or `to`...

@rijkvanzanten Anyways, I've just patched the PR, shifting logic out of the vue template. 

```js
export type SettingsModuleBarLinkBase = {
	type: 'link';
	id: string;
	icon: string;
	name: string;
	enabled: boolean;
	locked?: boolean;
};

export type SettingsModuleBarInternalLink = SettingsModuleBarLinkBase & {
	to: string;
	url?: string;
}

export type SettingsModuleBarExternalLink = SettingsModuleBarLinkBase & {
	url: string;
}

export type SettingsModuleBarLink = SettingsModuleBarInternalLink | SettingsModuleBarExternalLink;
```

Reply:
> That `href` vs `to question is only relevant at render time in that `module-bar.vue`component. We don't differentiate in the field itself in project-settings, so it's stored under the same`url` name regardless of how it has to be used in the end

Somehow when `module-bar` edits the `url` to `to`, it affects `system-module` which is using the data from props. But it's should be working now. 🤓